client/state/geolocation/actions.js

```diff
@@ -0,0 +1,32 @@
+import superagent from 'superagent';
+
+import {
+	GEOLOCATION_FETCH,
+	GEOLOCATION_FETCH_COMPLETED,
+	GEOLOCATION_FETCH_FAILED
+} from 'state/action-types';
+
+export function fetchGeolocation( dispatch ) {
+	return () => {
+		dispatch( {
+			type: GEOLOCATION_FETCH
+		} );
+
+		return new Promise( ( resolve, reject ) => {
+			superagent.get( 'https://public-api.wordpress.com/geo.php', null, ( error, res ) => {
```

Comment:
We recently (#6947, cc @retrofox ) upgraded `superagent` to 2.x, which [adds proper Promise support](https://github.com/visionmedia/superagent/blob/master/History.md#200), so instead of creating a Promise instance you could simply return this `get` call.